### PR TITLE
[Tests] Test async code generation in async tests.

### PR DESCRIPTION
**Pull request by erick-xanadu**
**Context:** Async tests were not testing that actual async code was generated.

**Description of the Change:** Adds code generation checks to async tests.

**Related GitHub Issues:** Uncovered #419

### CODE REVIEW

**Review by rmoyard — APPROVED** [review]
Thanks :+1:

**Comment by erick-xanadu** [comment]
I am going to wait until #424 is merged to merge this one. Thanks for the review!

**Pull request merged**
